components/input/Input.js

```diff
@@ -0,0 +1,196 @@
+import { TextSmall } from '../typography';
+import theme from './theme.css';
+
+export default class Input extends Component {
+  static propTypes = {
+    bold: PropTypes.bool,
+    className: PropTypes.string,
+    counter: PropTypes.number,
+    disabled: PropTypes.bool,
+    helpText: PropTypes.string,
+    icon: PropTypes.oneOfType([PropTypes.func, PropTypes.element]),
+    iconPlacement: PropTypes.oneOf(['left', 'right']),
+    id: PropTypes.string,
+    inverse: PropTypes.bool,
+    meta: InputMetaPropTypes,
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func,
+    onFocus: PropTypes.func,
+    placeholder: PropTypes.string,
+    readOnly: PropTypes.bool,
+    size: PropTypes.oneOf(['small', 'medium', 'large']),
+    step: PropTypes.number,
+    type: PropTypes.string,
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  };
+
+  static defaultProps = {
+    iconPlacement: 'left',
+    inverse: false,
+    disabled: false,
+    placeholder: '',
+    readOnly: false,
+    size: 'medium',
+    step: 1,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.isNumberInput = Boolean(props.type === 'number');
+    this.handleChange = ::this.onChange;
+    this.handleIncreaseValue = ::this.increaseValue;
+    this.handleDecreaseValue = ::this.decreaseValue;
+
+    this.state = {
+      value: this.isNumberInput ? Number(props.value) || '' : props.value || '',
+    };
+  }
+
+  componentWillUpdate(props, state) {
+    if (state.value) {
+      this.props.onChange(state.value);
+    }
+  }
+
+  onChange(event) {
+    const value = event.target.value;
+
+    this.setState(() => ({
+      value: this.isNumberInput ? Number(value) : value,
+    }));
+  }
+
+  increaseValue() {
+    this.setState((previousState, props) => ({
+      value: previousState.value ? previousState.value + props.step : props.step,
+    }));
+  }
+
+  decreaseValue() {
+    this.setState((previousState, props) => ({
+      value: previousState.value ? previousState.value - props.step : -props.step,
+    }));
+  }
+
+  renderInput() {
+    const { bold, disabled, id, onBlur, onFocus, placeholder, readOnly, step, type } = this.props;
+    const classNames = cx(theme['input'], {
+      [theme['is-bold']]: bold,
+    });
+    const props = {
+      className: classNames,
+      disabled: disabled,
+      id,
+      onBlur: onBlur,
+      onChange: this.handleChange,
+      onFocus: onFocus,
+      placeholder,
+      readOnly,
+      step,
+      type,
+      value: this.state.value,
+    };
+
+    return <input {...props} />;
+  }
+
+  renderCounter() {
+    if (this.props.counter) {
+      return <Counter className={theme.counter} count={this.props.counter} color="ruby" size="small" />;
+    }
+  }
+
+  renderHelpText() {
+    const { helpText, inverse } = this.props;
+
+    if (helpText) {
+      return (
+        <TextSmall color={inverse ? 'white' : 'neutral'} marginTop={1} soft>
+          {helpText}
+        </TextSmall>
```

Comment:
PP:

Reply:
PP: I also think here return is superfluous again, I might be wrong though because I didn't comment it like that and you still wrote 'return', could you confirm this please? @driesd

Reply:
Nevermind, it's in a function so idd you *do* have to return it :) 

 :ok: by @InstaK on 13/02/2018 12:08:54